python/dataset_plotting.py

```diff
@@ -0,0 +1,239 @@
+# Dataset imports
+from dataset import Data, dataset
+import numpy as np
+# Plotly imports
+from plotly.offline import init_notebook_mode, iplot
+init_notebook_mode(connected=True)
+import plotly.graph_objs as go
+
+#===============================================================================
+
+# Plot a 2D slice through a 3D dataset with a slider to adjust the position of
+# the slice in the third dimension
+def plot_sliceviewer(input_data, field=None):
+
+    # Delay import to here, as ipywidgets is not part of the base plotly package
+    try:
+        from ipywidgets import interactive, HBox, VBox
+    except ImportError:
+        print("Sorry, the sliceviewer requires ipywidgets which was not found on this system")
+        return
+
+    dims = input_data.dimensions()
+    if (len(dims) > 2) and (len(dims) < 5):
+
+        # Get spatial extents by iterating over the coordinates, and also search
+        # for data fields
+        axmin = []
+        axmax = []
+        axlab = []
+        fields = []
+        nx = []
+        ifield = 0
+        for var in input_data:
+            if var.is_coord:
+                values = var.numpy
+                axmin.append(values[0] - 0.5*(values[1]-values[0]))
+                axmax.append(values[-1] + 0.5*(values[-1]-values[-2]))
```

Comment:
Is this attempting to compute edges from center points? Why not just label the centers?

Reply:
This is to adjust the plotting area so that it doesn't clip off half of the outermost pixels